### DISCUSSION TITLE
ideas to make project more better

### DISCUSSION THREAD

**Discussion by SteveSuv** [excaliburjs/Excalibur#3034] [Ideas]
This is a good project, with better ts DX. But its influence is still small. I didn't know it until last week. I think these ideas can make it move forward fast and be better.
- turn multi repos to monorepo use [turborepo](https://turbo.build/repo/docs/getting-started/create-new) and [pnpm workspace](https://pnpm.io/workspaces)
   now this project has many repos, it's too scattered. monorepo can gather more attention in one project and is easier to maintain and expand. pnpm make packages management more efficient. this monorepo includes `engine`,`create-excalibur`,`site`,`plugins` four apps.
- move other plugin repos eg: ldtk, tiled to `plugins` app, then delete the origin repo, keep the only code entrance
- same with other template repos eg: parcel, webpack, vite, remove to `create-excalibur/templates`, then user type `pnpm create excalibur` can choose the template they want. but I think we can only use `vite`, `vite` has more and more became the only choice to dev with frontend project. others like parcel, webpack seems to be outdated. I think can keep templates like `vite` for web,`vite+electron` for desktop,`vite+react-native` for mobile. and remove other all templates repos.
- give up webpack, use [tsup](https://github.com/egoist/tsup) to build `engine` to esm/cjs/umd bundles
   webpack is so slow and tsup is faster, powerful and easy to config
- update `site` using [starlight](https://starlight.astro.build/) to replace Docusaurus 
   starlight's UI is more modernization and it's easier to maintain docs, will make project look more professional. 
   At the same time, we can also reorganize the document structure to make it clearer and novice-friendly.
   like we can update `examples` and `showcase` to show in codesanbox but not jump to github repos. it's more Intuitively
I think maybe it's worth to discuss how to make this project grow up faster to compete with more popular frameworks like phaser or kaboom.

**Comment by mattjennings** [comment]
What you are suggesting is no small effort. I'll speak for myself here, but the goal is not to become more popular than other frameworks like Phaser, and if it was, tooling is not the differentiator. These are all nice tools, I use some of these myself, but the benefits are negligible compared to the time and amount of changes needed.

> monorepo can gather more attention in one project and is easier to maintain and expand. pnpm make packages management more efficient

This is purely a matter of preference and, to be frank, a baseless claim. 

> give up webpack, use [tsup](https://github.com/egoist/tsup) to build engine to esm/cjs/umd bundles
> webpack is so slow and tsup is faster, powerful and easy to config

We already output esm/cjs/umd builds. Our build right now takes about 20 seconds on my macbook which is hardly an issue. If it were several minutes we'd probably look at using esbuild or vite or something.

> update site using [starlight](https://starlight.astro.build/) to replace Docusaurus
starlight's UI is more modernization and it's easier to maintain docs, will make project look more professional.

This is subjective. I use starlight myself, I like it, but Docusaurus is perfectly fine too.

**Comment by kamranayub** [comment]
I appreciate the suggestions. We may take a look at some of these if they become major pain points, but I don't think we're there yet. We just [completed a major migration to Docusaurus](https://github.com/excaliburjs/excaliburjs.github.io/issues/383), so I'm not too interested in another docs framework unless it truly adds some new value (I briefly looked at Starlight and didn't see a major differentiator). Theming/branding is a separate issue, and we'll eventually tackle that.

If people are truly interested in helping us grow the project, as  @mattjennings suggested, it's not _tooling_ but _marketing_ we would love help with.

Here are some concrete ways to help out as a community member:

- **Share your wins.** When people see you use Excalibur, they'll check us out. You don't need to be "market-y." Just share what you like and what it helps you with. This is how we get a lot of folks today. Share your wins wherever you hang out.
- **Teach other people Excalibur.** We are incredibly thankful for the AMAZING work Drew has done with his YT channel. If you feel comfortable showing people how to write games with Excalibur, and you have a platform, doing that also helps bring people into the fold. Doesn't matter if it's writing on dev.to, StackOverflow, your blog, or making YT videos -- all of it helps.
- **Publish your games!** Ultimately, we want people to make games... so the more games that get published, the more other developers will be aware we exist **and** it serves as "proof in the pudding" you can make production-level games.
- **Talk about it.** With your friends, colleagues, when someone asks for an engine recommendation, whatever -- right now, most of our growth is organic word-of-mouth. We now have [our Discord community](https://discord.gg/W6zUd4tTY3) to talk with other Excalibur devs and get help.

There are things we want to do as the core team, but ultimately, we are community-driven, so we lean on community members right now to grow.

> At the same time, we can also reorganize the document structure to make it clearer and novice-friendly. like we can update examples and showcase to show in codesanbox but not jump to github repos.

This is something I agree with, and we want to do (we're doing it... gradually). Docs are critical for our growth. We want to add more examples, we have the ability to embed live sandboxes (it's using Stackblitz), and we need to re-organize/fix the taxonomy of the docs to be more approachable. It's not easy because we have to maintain backward-compatibility with old docs permalinks, but we have some ideas on how to address that (either through Cloudflare rules or a Docusaurus redirect plugin).
